### PR TITLE
[JIT] Update freezing api

### PR DESCRIPTION
**Pull request by eellison**
Update freezing api  for 1.8,  and add a corresponding C++ API. The `optimize` flag hasn't been publicly released yet, so we are able to change it without breaking BC. I will submit a PR to branch release as well, there are a few more days to do that

### CODE REVIEW

File: torch/jit/_freeze.py [hunk 1]
```diff
@@ -140,4 +144,12 @@ def optimize_frozen_module(mod):
         assert "batch_norm" not in str(frozen_mod.graph)
 
     """
-    torch._C._jit_pass_optimize_frozen_graph(mod.graph)
+    # xxx: keep in sync with frozen_graph_optimization.cpp
+    # intentionally duplicated to make to make it easier to create custom optimization sequence
+    torch._C._jit_pass_remove_dropout(mod._c)
+    if optimize_numerics:
+        # run a couple times to capture Conv -> Mul -> Add etc
+        for _ in range(2):
```
Comment (dzhulgakov) [inline torch/jit/_freeze.py:152]: it's not very scientific, you probably want to iterate till the fixed point :) but it's unrelated to this diff

File: torch/jit/_freeze.py [hunk 2]
```diff
@@ -102,23 +100,29 @@ def forward(self, input):
...
     RecursiveScriptModule._finalize_scriptmodule(out)
-    if optimize:
-        optimize_frozen_module(out)
+    optimize_frozen_module(out, optimize_numerics)
 
     return out
 
 
-def optimize_frozen_module(mod):
+def optimize_frozen_module(mod, optimize_numerics: bool = True):
```
Comment (dzhulgakov) [inline torch/jit/_freeze.py:108]: if you get non-portable optimizations in the future - would it be a separate `bool` flag?

**Review by dzhulgakov — APPROVED** [review]
Looks reasonable, stamping to unblock you. @suo can chime in too

File: torch/jit/_freeze.py (hunk 2 above)
Comment (eellison) [reply torch/jit/_freeze.py:108]: yea, that was what i was envisioning. 

`def optimize_frozen_module(mod, optimize_numerics: bool = True, non_portable = False):`

File: torch/jit/_freeze.py (hunk 1 above)
Comment (eellison) [reply torch/jit/_freeze.py:152]: yea, that's kind of a todo, i'm not convinced it really matters but it would be a good follow up

**Pull request closed without merging**